src/common/lib/polkadot-provider/events/index.js

```diff
@@ -27,7 +27,7 @@ export async function processEvent(state, address, event, role) {
 
     const res = await handler(dataEvent, value, valueMessage)
     
-    if (res.data[identity] || res.data[1][identity] == address) {
+    if (res.data[identity] === address || res.data[1][identity] === address) {
```

Comment:
Why is `res.data[identity]` removed? Will it cause null errors?

Reply:
I don't think I got rid of it, I just fixed the condition because the `res.data[identity]` has no comparison like this `res.data[identity] === address`

Reply:
I think you should do it like this
`res.data[identity] || res.data[identity] === address || res.data[1][identity] === address)`

Please check that expression if it's correct or not.

Reply:
It's weird, have you tested this yet?

Reply:
yes, I have tested it. I think because the above condition has an await, it will not be continued when the request fails to execute. WDYT?

Reply:
Hmm, I am not so sure. Please make a unit test for this first.